### PR TITLE
Ensure packing job will continue run in case of error

### PR DESCRIPTION
**Pull request by kaloyan-raev**
So far an error in the packing job interrupted it altogether. Restarting the job required the manual restart of the node.

With this change the packing job continues to execute in the case of error. In addition, a monkit event is fired to register the error.

### CODE REVIEW

File: pack/chore.go [hunk 1]
```diff
@@ -88,9 +89,20 @@ func (chore *Chore) Run(ctx context.Context) {
 		chore.loop = sync2.NewCycle(chore.interval)
 
 		go func() {
```
Comment (egonelbre) [inline pack/chore.go:91]: Nothing is waiting for this goroutine to be completed. https://www.storj.io/blog/production-concurrency explains this problem further.

So there's a chance that it ends up using a closed `db.DB` or `*Store`.

File: pack/chore.go [hunk 2]
```diff
@@ -88,9 +89,20 @@ func (chore *Chore) Run(ctx context.Context) {
 		chore.loop = sync2.NewCycle(chore.interval)
```
Comment (egonelbre) [inline pack/chore.go:89]: Also `Run` implies that the function returns only when it has completely finished working. In this case, it's starting in a goroutine, so it should be called `Start` or you should avoid the goroutine in the first place.

**Review by egonelbre — APPROVED** [review]
I'm approving since it's an improvement on the existing code. The other comments can be addressed in a separate PR.

**Comment by kaloyan-raev** [comment]
@egonelbre Thank you! I will follow-up on your comments in a separate PR.

File: pack/chore.go (hunk 2 above)
Comment (kaloyan-raev) [reply pack/chore.go:89]: This is addressed in https://github.com/storj/ipfs-go-ds-storj/pull/68.

File: pack/chore.go (hunk 1 above)
Comment (egonelbre) [reply pack/chore.go:91]: We could try to do it over a google meet tomorrow.

**Pull request merged**